packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -931,10 +935,19 @@ def response_headers_and_deserialization(
                 )
             )
         elif response.type:
-            if self.code_model.options["models_mode"]:
+            if self.code_model.options["models_mode"] == "msrest":
                 retval.append(
                     f"deserialized = self._deserialize('{response.serialization_type}', pipeline_response)"
                 )
+            elif self.code_model.options["models_mode"] == "dpg":
+                deserializer = (
+                    None
+                    if not isinstance(response.type, ModelType)
+                    else f"_models.{response.type.name}"
+                )
+                retval.append(
+                    f"deserialized = _deserialize({deserializer}, response.json())"
```

Comment:
I think we had talked about sharing the code with `self._deserialize`, where we do a check if it's a dpg model or not. If it is, we do our own deserialization logic, otherwise we pass it to the vendored deserialization logic. This way the generated code is cleaner